File: Hotel/csharp/HOTELINTERFAZ/HOTELINTERFAZ/Models/Resena.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json.Serialization;

namespace HOTELINTERFAZ.Models
{
    public class ClienteReducido
    {
        [JsonPropertyName("_id")]
        public string Id { get; set; }

        [JsonPropertyName("dni")]
        public string Dni { get; set; }

        [JsonPropertyName("nombre")]
        public string Nombre { get; set; }
    }
    public class HabitacionReducida
    {
        [JsonPropertyName("numero")]
        public int Numero { get; set; }
    }

    public class Resena
    {
        [JsonPropertyName("_id")]
        public string Id { get; set; }

        [JsonPropertyName("clienteId")]
        public string ClienteId { get; set; }

        [JsonPropertyName("habitacionId")]
        public string HabitacionId { get; set; }

        [JsonPropertyName("reservaId")]
        public string ReservaId { get; set; }

        [JsonPropertyName("puntuacion")]
        public int Puntuacion { get; set; }

        [JsonPropertyName("comentario")]
        public string Comentario { get; set; }

        [JsonPropertyName("cliente")]
        public ClienteReducido Cliente { get; set; }

    }
}

File: Hotel/kotlin/app/src/main/java/com/example/aplicacion_hotel/Model/Resena.kt
package com.example.aplicacion_hotel.Model

import kotlinx.serialization.Serializable

@Serializable
data class Resena(
    val _id: String? = null,
    val clienteId: String,
    val reservaId: String,
    val habitacionId: String,
    val comentario: String,
    val puntuacion: Int,
    val fecha: String? = null
)

@Serializable
data class CrearResenaRequest(
    val clienteId: String,
    val reservaId: String,
    val habitacionId: String,
    val comentario: String,
    val puntuacion: Int
)
